pkg/aws/aws_suite_test.go

```diff
@@ -32,15 +32,18 @@ const (
 	testNamespace = "test"
 )
 
+//params is used to pass annotations to the machine spec
 func newMachine(
-	setMachineIndex int,
+	setMachineIndex int, params ...interface{},
```

Comment:
```suggestion
	setMachineIndex int, annotations map[string]string,
```
I would have expected this method to be simpler like this. This would help future contributors to easily understand the expected parameters. I know the interface would provide the flexibility to add any type/number of params, however, I feel it would be difficult for new contributors to understand. 

However, if you have strong arguments for having interfaces please let me know as I am not an expert myself and may have overlooked the advantages.

Reply:
@prashanth26 Actually, this was my idea/suggestion to Himanshu to introduce the variadic parameter of type empty interface exactly for the reason as you have inferred `the interface would provide the flexibility to add any type/number of params`. So probably, I will explain the reasoning.

The reason I thought of this approach is that any new parameter addition to the function `newMachine()` by a developer in future will not force the developer to modify all the function calls at other places.

However, I missed thinking about the fact that you mentioned
>I feel it would be difficult for new contributors to understand

I second your suggestion here 😄 Thank you!

Reply:
> The reason I thought of this approach is that any new parameter addition to the function newMachine() by a developer in future will not force the developer to modify all the function calls at other places.

I agree with you as well. I thought about this after I made my comment. In some sense, I guess both have it's advantages and disadvantages. 

Himanshu - If you feel the change to annotations as suggested above it's too tiring/difficult to make. I think a simple `ctrl+f` replace `newMachine(<regexp>, <number>)` to `newMachine(<regexp>, <number>, nil)` should do the trick to fix other invokations to this method. However, If you find it difficult I think the current approach is also okay. However, we should improve the comments with @AxiomSamarth 's suggestion above so that new contributors aren't confused. :)

Reply:
I just looked up for some best practices with respect to using empty interfaces and below are the highlights/findings-

- `interfaces{}` should be the last thing to be used when there is no other straight forward way because using `interface{}` would also carry a burden of introducing type assertions which it already has [here](https://github.com/himanshu-kun/machine-controller-manager-provider-aws/blob/9e8f0cdf154c50bd7f80ad427894c01d852d3788/pkg/aws/aws_suite_test.go#L61) (which in turn may carry some programmatic limitations).

In this situation, introducing apt parameter of type `map[string]string` looks like a better option to me. Again, even I shall leave it to Himashu's decision 😄

Reply:
> I just looked up for some best practices with respect to using empty interfaces and below are the highlights/findings-
> 
> * `interfaces{}` should be the last thing to be used when there is no other straight forward way because using `interface{}` would also carry a burden of introducing type assertions which it already has [here](https://github.com/himanshu-kun/machine-controller-manager-provider-aws/blob/9e8f0cdf154c50bd7f80ad427894c01d852d3788/pkg/aws/aws_suite_test.go#L61) (which in turn may carry some programmatic limitations).
> 
> In this situation, introducing apt parameter of type `map[string]string` looks like a better option to me. Again, even I shall leave it to Himashu's decision 😄

I would also prefer the `map[string]string` param approach. I would think it's less confusing to new developers. Let's do with this then.

Reply:
I am thinking of changing `newMachine(setMachineIndex int, params ...interface{})` to `newMachine(setMachineIndex int, params ...map[string]string)`
also I didn't get the `ctrl+f` approach mentioned by @prashanth26

Reply:
>I am thinking of changing newMachine(setMachineIndex int, params ...interface{}) to newMachine(setMachineIndex int, params ...map[string]string)

Yes, that is the only change required.

> I didn't get the ctrl+f approach mentioned by @prashanth26

This was just a trick for you to update existing method invocations to the new signature style.